include/identity.php

```diff
@@ -50,13 +50,25 @@ function profile_load(App $a, $nickname, $profile = 0, $profiledata = array(), $
 		dbesc($nickname)
 	);
 
-	if (!$user && count($user) && !count($profiledata)) {
+	if (!$user && !count($user) && !count($profiledata)) {
 		logger('profile error: ' . $a->query_string, LOGGER_DEBUG);
 		notice(t('Requested account is not available.') . EOL);
 		$a->error = 404;
 		return;
 	}
 
+	if (!x($a->page, 'aside')) {
+		$a->page['aside'] = '';
+	}
+
+	if ($profiledata) {
```

Comment:
This function is often called with $profiledata populated but an empty nickname. In this case, all it does is output a profile aside with the provided `$profiledata` since no user can be fetched with an empty `$nickname`. But in the meantime it's lighting up the error log because we don't check for validity of `$user` at each step.

Reply:
I guess you checked that all variables are valid at this point and don't get modified after it?

Reply:
Yeah, this function should have been split in two long ago.

Reply:
It was in the dark old days where we added functionality without much thoughts.